cmd/promtool/unittest.go

```diff
@@ -323,11 +325,47 @@ func (tg *testGroup) test(evalInterval time.Duration, groupOrderMap map[string]i
 					}
 					expString := indentLines(expAlerts.String(), "            ")
 					gotString := indentLines(gotAlerts.String(), "            ")
-					errs = append(errs, fmt.Errorf("%s    alertname: %s, time: %s, \n        exp:%v, \n        got:%v",
-						testName, testcase.Alertname, testcase.EvalTime.String(), expString, gotString))
+
+					// If empty, populates an empty value
+					if gotAlerts.Len() == 0 {
+						gotAlerts = append(gotAlerts, labelAndAnnotation{
+							Labels:      labels.Labels{},
+							Annotations: labels.Labels{},
+						})
+					}
+					// If empty, populates an empty value
+					if expAlerts.Len() == 0 {
+						expAlerts = append(expAlerts, labelAndAnnotation{
+							Labels:      labels.Labels{},
+							Annotations: labels.Labels{},
+						})
+					}
+
+					diffOpts := jsondiff.DefaultConsoleOptions()
+					expAlertsJSON, err := json.Marshal(expAlerts)
+					if err != nil {
+						errs = append(errs, fmt.Errorf("alertName: %s\nError converting expAlert to JSON: %v", tg.TestGroupName, expAlerts.String()))
+						continue
+					}
+
+					gotAlertsJSON, err := json.Marshal(gotAlerts)
+					if err != nil {
+						errs = append(errs, fmt.Errorf("alertName: %s\nError converting gotAlert to JSON: %v", tg.TestGroupName, gotAlerts.String()))
+						continue
+					}
+
+					res, diff := jsondiff.Compare(expAlertsJSON, gotAlertsJSON, &diffOpts)
+					if res != jsondiff.FullMatch {
+						if runtime.GOOS == "windows" {
```

Comment:
Rather than doing this based on windows or not it would be nicer to have a --color=always/auto/never option. (Like `ls`)

Windows thesedays can support VT colors, it might need something like: https://github.com/moby/term/blob/master/term_windows.go#L52 to enable it. (See https://learn.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences)

Reply:
I tested on vscode terminal in windows & the colors seem to work. They are just not working on the default cmd & I think that's okay.